iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -2157,7 +2180,7 @@ private Partition getPartitionImpl(org.apache.hadoop.hive.ql.metadata.Table tabl
   public List<String> getPartitionNames(org.apache.hadoop.hive.ql.metadata.Table hmsTable,
       Map<String, String> partitionSpec) throws SemanticException {
     Table icebergTable = IcebergTableUtil.getTable(conf, hmsTable.getTTable());
-    return IcebergTableUtil.getPartitionNames(icebergTable, partitionSpec, true);
+    return IcebergTableUtil.getPartitionNames(icebergTable, partitionSpec, false);
```

Comment:
This method is called from a few locations, like `ShowPartitionsOperation`, does it work correctly with the change to include partitions for old specs? Should `ShowPartitionsOperation` show partitions for older specs?

Reply:
yes, [get/show]Partitions by default should return ALL table partitions